### PR TITLE
charts/csm-application-mobility Added Application Mobility helm chart

### PR DESCRIPTION
**Pull request by tdawe**
#### Is this a new chart?

Yes

#### What this PR does / why we need it:
This PR contains a new helm chart for CSM Application Mobility.

#### Which issue(s) is this PR associated with:

- https://github.com/dell/csm/issues/449

#### Special notes for your reviewer:
Do not merge until RFD (September 21, 2022)

#### Checklist:

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Chart Version bumped
- [x] Variables are documented in the chart README.md (**Note: Added to official CSM Documentation**)
- [x] Title of the PR starts with the chart name (e.g. `[charts_dir/mychartname]`) if applicable

### CODE REVIEW

**Review by bpjain2004 — APPROVED** [review]

File: charts/csm-application-mobility/values.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Default values for app-mobility.
```
Comment (sharmilarama) [inline charts/csm-application-mobility/values.yaml:1]: Comments here should be renamed as well.

File: charts/csm-application-mobility/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,122 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "app-mobility.name" -}}
```
Comment (sharmilarama) [inline charts/csm-application-mobility/templates/_helpers.tpl:4]: Should the references here be renamed as well?

File: charts/csm-application-mobility/values.yaml (hunk 1 above)
Comment (tdawe) [reply charts/csm-application-mobility/values.yaml]: Fixed the comments to be consistent with name of the chart

File: charts/csm-application-mobility/templates/_helpers.tpl (hunk 2 above)
Comment (tdawe) [reply charts/csm-application-mobility/templates/_helpers.tpl]: Renamed these to be consistent. Verified that the same yaml is generated during `helm install` after renaming these to csm-application-mobility.

**Review by bpjain2004 — DISMISSED** [review]

**Review by sharmilarama — APPROVED** [review]
LGTM

**Review by alikdell — APPROVED** [review]

**Review by atye — APPROVED** [review]

**Pull request merged**